indico/modules/receipts/util.py

```diff
@@ -145,19 +145,33 @@ def compile_jinja_code(code: str, template_context: dict, *, use_stack: bool = F
         raise UnprocessableEntity(e)
 
 
-def sandboxed_url_fetcher(event: Event, allow_event_logo: bool = False) -> t.Callable[[str], dict]:
+def sandboxed_url_fetcher(event: Event, allow_event_images: bool = False) -> t.Callable[[str], dict]:
     """Fetch also "event-local" URLs.
 
     More info on fetchers: https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#url-fetchers
     """
     allow_external_urls = receipts_settings.get('allow_external_urls')
 
     def _fetcher(url: str) -> dict:
-        if allow_event_logo and url == 'event://logo':
-            return {
-                'mime_type': event.logo_metadata['content_type'],
-                'string': event.logo
-            }
+        if allow_event_images and url.startswith('event://'):
+            img_ref = url[8:]
+            if img_ref == 'logo':
+                return {
+                    'mime_type': event.logo_metadata['content_type'],
+                    'string': event.logo
+                }
+            elif img_ref.startswith('layout_images/'):
+                try:
+                    image = event.layout_images.filter_by(id=img_ref[14:]).one()
+                except Exception:
```

Comment:
we don't do that.... ;) catch the actual exception or use `.first()` and `if not ...`